lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -417,9 +543,31 @@ def save_element_message(
         """
         id_to_save = self.select_dg_to_save(invoked_dg_id, used_dg_id)
         for msgs in self._cached_message_stack:
-            msgs.append(
-                ElementMsgData(delta_type, element_proto, id_to_save, returned_dg_id)
-            )
+            try:
+                wid = element_proto.id  # type: ignore
```

Comment:
~Why `# type: ignore`?~ Edit: oh, I see - it's because it's just the protobuf message base class.

Could we define and use a `Protocol` here to achieve type safety?